### PR TITLE
Upgrade linux CI from trusty to xenial

### PR DESCRIPTION
**Pull request by jdm**
I'm going to upgrade all the machines, deploy the changes in this branch, and we can merge when the process is done.

I have successfully completed builds on both a new linux builder and a new linux cross builder that were upgraded from trusty.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/saltfs/957)

### CODE REVIEW

**Comment by jdm** [comment]
* [ ] servo-linux1
* [ ] servo-linux2
* [ ] servo-linux3
* [ ] servo-linux4
* [ ] servo-linux5
* [ ] servo-linux6
* [ ] servo-linux-cross1
* [ ] servo-linux-cross2
* [ ] servo-linux-cross3

**Comment by jdm** [comment]
Current status: all common update steps performed on all machines; highstate in progress.

**Comment by jdm** [comment]
I just noticed all of the linux bustage on travis, which is worrying!

**Comment by jdm** [comment]
Based on https://www.reddit.com/r/saltstack/comments/ae85xg/installingupdating_python_module/, it may be possible to downgrade the version of pip on TravisCI builds to avoid this error:
```
----------
          ID: buildbot-slave-dependencies
    Function: pip.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1905, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1830, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/pip_state.py", line 681, in installed
                  out = _check_pkg_version_format(pkg)
                File "/usr/lib/python2.7/dist-packages/salt/states/pip_state.py", line 147, in _check_pkg_version_format
                  install_req = InstallRequirement.from_line(pkg)
              AttributeError: type object 'InstallRequirement' has no attribute 'from_line'
```
That leaves these PPA issues, which don't appear on our builders because the files are already present during the upgrade:
```
----------
          ID: /etc/apt/sources.list.d/openjdk-r-ppa-trusty.list
    Function: file.exists
      Result: False
     Comment: Specified path /etc/apt/sources.list.d/openjdk-r-ppa-trusty.list does not exist
     Started: 15:50:23.105446
    Duration: 0.825 ms
     Changes:   
----------
          ID: cmake-ppa
    Function: pkgrepo.managed
      Result: False
     Comment: Failed to configure repo 'cmake-ppa': Unable to add PPA 'george-edison55/cmake-3.x'. '[u'apt-add-repository', u'-y', u'ppa:george-edison55/cmake-3.x']' exited with status 1: 'gpg: keyring `/tmp/tmpuj417wiv/secring.gpg' created
              gpg: keyring `/tmp/tmpuj417wiv/pubring.gpg' created
              gpg: requesting key 828AB726 from hkp server keyserver.ubuntu.com
              gpg: no valid OpenPGP data found.
              gpg: Total number processed: 0
              gpg: keyserver communications error: keyserver helper general error
              gpg: keyserver communications error: unknown pubkey algorithm
              gpg: keyserver receive failed: unknown pubkey algorithm'
----------
          ID: ffmpeg-ppa
    Function: pkgrepo.managed
      Result: False
     Comment: Failed to configure repo 'ffmpeg-ppa': Unable to add PPA 'jonathonf/ffmpeg-3'. '[u'apt-add-repository', u'-y', u'ppa:jonathonf/ffmpeg-3']' exited with status 1: 'gpg: keyring `/tmp/tmpwd7blizj/secring.gpg' created
              gpg: keyring `/tmp/tmpwd7blizj/pubring.gpg' created
              gpg: requesting key F06FC659 from hkp server keyserver.ubuntu.com
              gpg: no valid OpenPGP data found.
              gpg: Total number processed: 0
              gpg: keyserver communications error: keyserver helper general error
              gpg: keyserver communications error: unknown pubkey algorithm
              gpg: keyserver receive failed: unknown pubkey algorithm'
----------
          ID: ttf-mscorefonts-installer
    Function: debconf.set
      Result: False
     Comment: State 'debconf.set' was not found in SLS 'servo-build-dependencies'
              Reason: 'debconf.set' is not available.
     Changes:   
```

**Comment by jdm** [comment]
It's greeeeeeeeeeeen!

**Comment by jdm** [comment]
The salt upgrade works around https://github.com/saltstack/salt/issues/49967 which was fixed in 2018.3.4. The python upgrade works around https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings.

File: xvfb/xvfb.service [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Job that runs xvfb
+Documentation=man:Xvfb(1)
+
+[Service]
+ExecStart=/usr/bin/Xvfb :0 -screen 0 1280x960x24 +extension RANDR +extension RENDER +extension GLX
+
+[Install]
+WantedBy=multi-user.target
```
Comment (Manishearth) [inline xvfb/xvfb.service]: newline at eof?

**Pull request merged**